src/backend/distributed/metadata/distobject.c

```diff
@@ -109,13 +109,20 @@ citus_unmark_object_distributed(PG_FUNCTION_ARGS)
 	Oid classid = PG_GETARG_OID(0);
 	Oid objid = PG_GETARG_OID(1);
 	int32 objsubid = PG_GETARG_INT32(2);
+
+	/*
+	 * SQL function master_unmark_object_distributed doesn't expect the
+	 * 4th argument but SQL function citus_unmark_object_distributed does
+	 * so as checkobjectexistence argument. For this reason, we try to
+	 * get the 4th argument only if this C function is called with 4
+	 * arguments.
+	 */
 	bool checkObjectExistence = true;
-	if (!PG_ARGISNULL(3))
+	if (PG_NARGS() == 4)
```

Comment:
A note for reviewers;

PG_ARGISNULL() was not the correct way to check for above case. This is because;
1) This is a "strict" function, so this means that no arguments can be NULL at the run-time. In other words, when any of the arguments is provided as NULL, then Postgres doesn't even execute the function but returns a NULL value.
2) The correct way to check presence of an argument is to use PG_NARGS().